### PR TITLE
Framebuffer and lcddev buffer usage for LCD displays drivers

### PR DESCRIPTION
**Pull request by tmedicci**
## Summary
PR following the discussion in https://github.com/apache/incubator-nuttx/pull/6657
In summary, LCD device driver and framebuffer are not able to use both the same `putarea` implementations of the LCD driver.
General picture is also described at the first commit description.

## Impact
This impacts the way LCD displays refresh an area. 

## Testing
Tested with both LVGL simulator config (LCD and FB) as well as with the TTGO T-Display board (which contains a st7789 TFT display).

### CODE REVIEW

File: include/nuttx/lcd/lcd.h [hunk 1]
```diff
@@ -90,7 +92,8 @@ struct lcd_planeinfo_s
 
   int (*putarea)(FAR struct lcd_dev_s *dev, fb_coord_t row_start,
                  fb_coord_t row_end, fb_coord_t col_start,
-                 fb_coord_t col_end, FAR const uint8_t *buffer);
+                 fb_coord_t col_end, FAR const uint8_t *buffer,
+                 fb_coord_t stride);
```
Comment (xiaoxiang781216) [inline include/nuttx/lcd/lcd.h:96]: let add the same parameter to get_area too?

Comment (tmedicci) [reply include/nuttx/lcd/lcd.h:96]: Yes, it makes total sense! I'll update the PR, thanks!

Comment (tmedicci) [reply include/nuttx/lcd/lcd.h:96]: Updated it ;)

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM!

**Comment by tmedicci** [comment]
need to fix a small bugfix for gc9a01 LCD driver. Just a minute.

**Comment by tmedicci** [comment]
> need to fix a small bugfix for gc9a01 LCD driver. Just a minute.

Done ;)

**Comment by tmedicci** [comment]
@acassis , if possible, can you test this with the APA102-based LCD matrix?

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by lupyuen** [comment]
Thanks @tmedicci! This works great on the ST7789 Display for PineDio Stack BL604. I have changed `LCDDEVIO_PUTRUN` back to `LCDDEVIO_PUTAREA` in my LVGL demo: https://github.com/apache/incubator-nuttx/pull/6657#issuecomment-1204657460

**Pull request merged**